presto-spi/src/main/java/com/facebook/presto/spi/statistics/Range.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import java.util.Objects;
+
+import static java.lang.Double.isNaN;
+import static java.lang.String.format;
+
+public class Range
+{
+    private final double min;
+    private final double max;
+
+    public Range(double min, double max)
+    {
+        if (isNaN(min)) {
+            throw new IllegalArgumentException("min must not be NaN");
+        }
+        if (isNaN(max)) {
+            throw new IllegalArgumentException("max must not be NaN");
+        }
+        if (max < min) {
+            throw new IllegalArgumentException(format("max must be greater than or equal to min. max: %s. min: %s.", max, min));
+        }
+        this.min = min;
+        this.max = max;
+    }
+
+    public double getMin()
+    {
+        return min;
+    }
+
+    public double getMax()
+    {
+        return max;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Range range = (Range) o;
+        return Double.compare(range.min, min) == 0 &&
+                Double.compare(range.max, max) == 0;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(min, max);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Range{" +
```

Comment:
@mbasmanova in SPI we don't have Guava dependency (and don't want to have it)

Reply:
@findepi Piotr, thanks for explaining. In this case, let's copy-paste most useful utilities like this one.

Reply:
@mbasmanova  We didn't do this yet, because we didn't want to add publicly accessible classes not intended for public consumption and package private didn't make it, as there as subpackages in SPI...  Of course, this is something we can revisit anytime. I am all for it.

Reply:
This is something we can explore after moving to Java 9 modules, as that will allow us to have module-private cross-package utility code.